modules/features/settings/src/main/java/au/com/shiftyjelly/pocketcasts/settings/history/upnext/UpNextHistoryDetailsPage.kt

```diff
@@ -0,0 +1,245 @@
+            actions = { iconColor ->
+                TextButton(
+                    onClick = onRestoreClick,
+                    enabled = state is UiState.Loaded,
+                ) {
+                    TextH50(text = stringResource(LR.string.restore))
+                }
+            },
+        )
+        when (state) {
+            is UiState.Loading -> LoadingView()
+            is UiState.Loaded -> {
+                UpNextHistoryEpisodes(
+                    state = state,
+                    bottomInset = bottomInset,
+                )
+            }
+
+            is UiState.Error -> UpNextHistoryErrorView()
+        }
+    }
+}
+
+@Composable
+private fun UpNextHistoryEpisodes(
+    state: UiState.Loaded,
+    bottomInset: Dp,
+    modifier: Modifier = Modifier,
+) {
+    LazyColumn(
+        modifier
+            .fillMaxHeight()
+            .padding(16.dp),
+        contentPadding = PaddingValues(bottom = bottomInset),
+    ) {
+        items(state.episodes) { episode ->
+            EpisodeRow(
+                episode = episode,
+                useEpisodeArtwork = state.useEpisodeArtwork,
+            )
+        }
+    }
+}
+
+@Composable
+private fun UpNextHistoryErrorView() {
+    Box(
+        modifier = Modifier.fillMaxSize(),
+        contentAlignment = Alignment.Center,
+    ) {
+        TextH50(
+            text = stringResource(LR.string.error_generic_message),
+            modifier = Modifier.padding(horizontal = 16.dp),
+        )
+    }
+}
+
+@Composable
+private fun EpisodeRow(
+    episode: BaseEpisode,
+    useEpisodeArtwork: Boolean,
+    modifier: Modifier = Modifier,
+) {
+    val context = LocalContext.current
+    val formattedDuration = remember(episode.durationMs) {
+        TimeHelper.getTimeDurationMediumString(
+            episode.durationMs,
+            context,
+        )
+    }
+    val published = remember(episode.durationMs) {
+        val dateFormatter = RelativeDateFormatter(context)
+        dateFormatter.format(episode.publishedDate)
+    }
+
+    Row(
+        verticalAlignment = Alignment.CenterVertically,
+        modifier = Modifier
+            .fillMaxWidth()
+            .padding(vertical = 4.dp),
+    ) {
+        EpisodeImage(
+            episode = episode,
+            useEpisodeArtwork = useEpisodeArtwork,
+            modifier = modifier.size(56.dp),
+        )
+
+        Column(
+            modifier = Modifier
+                .weight(1f)
+                .padding(start = 16.dp),
+        ) {
+            TextH70(
+                text = published,
+                maxLines = 1,
+                modifier = Modifier.padding(bottom = 4.dp),
+            )
+
+            TextH40(
+                text = episode.title,
+                maxLines = 1,
+            )
+
+            TextH70(
+                text = formattedDuration,
+                maxLines = 1,
+                modifier = Modifier.padding(top = 4.dp),
+            )
```

Comment:
Can we make these look more like the episode row from filters?

| | |
| --- | --- |
| <img width="317" alt="Screenshot 2025-03-21 at 11 04 59 am" src="https://github.com/user-attachments/assets/38b7f3eb-57e7-4f71-8bd1-728b7b411454" /> | <img width="288" alt="Screenshot 2025-03-21 at 11 03 51 am" src="https://github.com/user-attachments/assets/a3a55879-f143-4ac7-b166-ba3960523fc7" /> |

Reply:
Updated in 3980833

<img width = 300  src="https://github.com/user-attachments/assets/6a6a21dc-c425-4625-9533-bde6f12ef625"/>